### PR TITLE
K8SPG-75 Add users pass handling

### PR DESCRIPTION
**Pull request by nonemax**
This PR add possibility to manage system users passwords via k8s secret

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| scheduled-backup | passed |
| init-deploy | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| upgrade | passed |
| users | passed |
| data-migration-gcs | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/fda8dce9867844e656c35acabf9cd340fdee9218
image: `perconalab/percona-postgresql-operator:PR-146-fda8dce9-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-fda8dce9-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-fda8dce9-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-fda8dce9-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-fda8dce9-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-fda8dce9-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| scaling | failed |
| recreate | failed |
| affinity | failed |
| monitoring | failed |
| self-healing | failed |
| operator-self-healing | failed |
| clone-cluster | failed |
| tls-check | failed |
| upgrade | passed |
| users | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/87c36681e8bc276c4d9f58bf5792461fb02bf72b
image: `perconalab/percona-postgresql-operator:PR-146-87c36681-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-87c36681-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-87c36681-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-87c36681-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-87c36681-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-87c36681-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |
| clone-cluster | passed |
| tls-check | passed |
| upgrade | passed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/6a673aef147dfbe59cee18402b84d0cf36503104
image: `perconalab/percona-postgresql-operator:PR-146-6a673aef-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-6a673aef-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-6a673aef-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-6a673aef-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-6a673aef-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-6a673aef-pgo-deployer`

**Review by egegunes — APPROVED** [review]

File: pkg/apis/crunchydata.com/v1/percona-cluster.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ type PerconaPGClusterSpec struct {
 	SSLCA                    string                 `json:"sslCA"`
 	SSLSecretName            string                 `json:"sslSecretName"`
 	SSLReplicationSecretName string                 `json:"sslReplicationSecretName"`
+	UsersSecretName          string                 `json:"usersSecretName"`
```
Comment (hors) [inline pkg/apis/crunchydata.com/v1/percona-cluster.go:47]:
```suggestion
	UsersSecretName          string                 `json:"secretsName"`
```

File: percona/controllers/pgc/secrets.go [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+	if len(password) == 0 {
+		generatedPassword, err := util.GeneratePassword(util.DefaultGeneratedPasswordLength)
+		if err != nil {
+			return SecretData{}, errors.Wrap(err, "generate password")
+		}
+		password = generatedPassword
+	}
+
+	switch userName {
+	case "postgres":
```
Comment (hors) [inline percona/controllers/pgc/secrets.go:142]: also we need to have possibility of changing password for  user which was created from CR https://github.com/percona/percona-postgresql-operator/blob/main/deploy/cr.yaml#L17

**Review by hors — CHANGES_REQUESTED** [review]

File: e2e-tests/users/run [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+deploy_operator
+cluster="some-name"
+kubectl_bin apply -f $conf_dir/client.yml
+
+spinup_pgcluster "${cluster}" "${src_dir}/deploy/cr.yaml"
+compare_psql_cmd "show-1" 'SHOW DATABASES' "pgbouncer:pgbouncer_pass@${cluster}-pgbouncer.${namespace}"
+compare_psql_cmd "select-1" '\c myapp \\\ SELECT * from myApp;' "postgres:postgres_pass@${cluster}.${namespace}"
+
+newpass="test-password"
+newpassencrypted=$(echo -n "$newpass" | base64)
```
Comment (cap1984) [inline e2e-tests/users/run:19]: please add here some regular `some-name` user without root privileges

File: deploy/users-secret.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+data:
+  pgbouncer: bmV3X3Bhc3N3b3JkXzE=
+  postgres: bmV3X3Bhc3N3b3JkXzE=
+  primaryuser: bmV3X3Bhc3N3b3JkXzE=
+kind: Secret
```
Comment (cap1984) [inline deploy/users-secret.yaml:6]: would user secrets creation be still enabled by default if `cluste1-users` is missing?

File: percona/controllers/pgc/pgc.go [hunk 5]
```diff
@@ -216,6 +224,42 @@ func (c *Controller) handleStatuses() error {
 	return nil
 }
 
+func (c *Controller) reconcileUsers(cluster *crv1.PerconaPGCluster) error {
```
[the remark this answers is not in the document]
Comment (nonemax) [reply percona/controllers/pgc/pgc.go:227]: we already do this

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| upgrade | passed |
| users | passed |
| scheduled-backup | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/a801eaf62403920d715a79466d369289e7ee812d
image: `perconalab/percona-postgresql-operator:PR-146-a801eaf6-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-a801eaf6-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-a801eaf6-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-a801eaf6-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-a801eaf6-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-a801eaf6-pgo-deployer`

File: percona/controllers/pgc/users.go [hunk 6]
```diff
@@ -19,7 +19,7 @@ import (
 )
 
 const (
-	sqlAlterRole      = `ALTER ROLE %s`
+	sqlAlterRole      = `ALTER USER %s`
```
Comment (cap1984) [inline percona/controllers/pgc/users.go:22]: so, sqlAlterRole or sqlAlterUser

File: percona/controllers/pgc/users.go [hunk 7]
```diff
@@ -36,6 +36,7 @@ func (c *Controller) UpdateUsers(usersSecret *v1.Secret, clusterName, namespace
 }
 
 func updateUserPassword(clientset kubeapi.Interface, username, password string, cluster *crv1.Pgcluster) error {
+	fmt.Println("Update user password for", username, password)
```
Comment (cap1984) [inline percona/controllers/pgc/users.go:39]: plain text password in the operator output?

Comment (nonemax) [reply percona/controllers/pgc/users.go]: removed, thanks https://github.com/percona/percona-postgresql-operator/pull/146/commits/0ad14b7f1c2cc837914914116dc3ec177514d8a8

File: percona/controllers/pgc/users.go (hunk 6 above)
Comment (nonemax) [reply percona/controllers/pgc/users.go]: https://github.com/percona/percona-postgresql-operator/pull/146/commits/0ad14b7f1c2cc837914914116dc3ec177514d8a8

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| upgrade | passed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/0ad14b7f1c2cc837914914116dc3ec177514d8a8
image: `perconalab/percona-postgresql-operator:PR-146-0ad14b7f-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-0ad14b7f-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-0ad14b7f-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-0ad14b7f-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-0ad14b7f-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-0ad14b7f-pgo-deployer`

File: e2e-tests/users/run (hunk 3 above)
Comment (nonemax) [reply e2e-tests/users/run:19]: added in https://github.com/percona/percona-postgresql-operator/commit/0ad14b7f1c2cc837914914116dc3ec177514d8a8

File: deploy/users-secret.yaml (hunk 4 above)
Comment (nonemax) [reply deploy/users-secret.yaml:6]: secret will be generated automatically

**Review by cap1984 — APPROVED** [review]

File: percona/controllers/pgc/pgc.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (hors) [reply percona/controllers/pgc/pgc.go:227]: @nonemax Are you sure ?
```
❯ kubectl get secrets cluster1-users
NAME             TYPE     DATA   AGE
cluster1-users   Opaque   3      6m21s
❯ kubectl get pvc
No resources found in pgo namespace.
```

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| scheduled-backup | failed |
| clone-cluster | passed |
| tls-check | passed |
| upgrade | passed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/95f3fb6bef1ead16a9b0ba4a04bccb61b81a6e7e
image: `perconalab/percona-postgresql-operator:PR-146-95f3fb6b-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-95f3fb6b-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-95f3fb6b-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-95f3fb6b-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-95f3fb6b-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-95f3fb6b-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| scheduled-backup | passed |
| init-deploy | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| upgrade | passed |
| users | passed |
| data-migration-gcs | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/95f3fb6bef1ead16a9b0ba4a04bccb61b81a6e7e
image: `perconalab/percona-postgresql-operator:PR-146-95f3fb6b-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-95f3fb6b-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-95f3fb6b-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-95f3fb6b-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-95f3fb6b-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-95f3fb6b-pgo-deployer`

File: percona/controllers/pgc/pgc.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (nonemax) [reply percona/controllers/pgc/pgc.go:227]: https://github.com/percona/percona-postgresql-operator/pull/146/commits/95f3fb6bef1ead16a9b0ba4a04bccb61b81a6e7e

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |
| clone-cluster | passed |
| tls-check | failed |
| upgrade | failed |
| users | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/1af1a590d4fd5c99ddea64ac34ae83b4022883f5
image: `perconalab/percona-postgresql-operator:PR-146-1af1a590-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-1af1a590-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-1af1a590-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-1af1a590-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-1af1a590-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-1af1a590-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | failed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| scheduled-backup | failed |
| recreate | passed |
| affinity | failed |
| monitoring | passed |
| self-healing | failed |
| operator-self-healing | failed |
| clone-cluster | failed |
| tls-check | failed |
| upgrade | failed |
| users | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/ec860939dc0df377e20135e1b6122a93726568fd
image: `perconalab/percona-postgresql-operator:PR-146-ec860939-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-ec860939-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-ec860939-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-ec860939-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-ec860939-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-ec860939-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | failed |
| init-deploy | failed |
| data-migration-gcs | failed |
| scaling | failed |
| scheduled-backup | failed |
| recreate | failed |
| affinity | failed |
| monitoring | failed |
| self-healing | failed |
| operator-self-healing | failed |
| clone-cluster | failed |
| tls-check | failed |
| upgrade | failed |
| users | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/853173b721a031e237210d1a8d366c96934cc1d0
image: `perconalab/percona-postgresql-operator:PR-146-853173b7-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-853173b7-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-853173b7-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-853173b7-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-853173b7-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-853173b7-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | failed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| scheduled-backup | passed |
| tls-check | failed |
| upgrade | failed |
| users | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/37eecd082e3b831db9f2f580d4146314c33f2db5
image: `perconalab/percona-postgresql-operator:PR-146-37eecd08-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-37eecd08-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-37eecd08-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-37eecd08-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-37eecd08-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-37eecd08-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | failed |
| data-migration-gcs | passed |
| upgrade | failed |
| users | failed |
| scheduled-backup | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/37eecd082e3b831db9f2f580d4146314c33f2db5
image: `perconalab/percona-postgresql-operator:PR-146-37eecd08-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-37eecd08-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-37eecd08-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-37eecd08-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-37eecd08-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-37eecd08-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| scheduled-backup | failed |
| clone-cluster | passed |
| tls-check | failed |
| upgrade | failed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/c4b57c4e95e9ebf29f484686978e61281a4bd3df
image: `perconalab/percona-postgresql-operator:PR-146-c4b57c4e-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-c4b57c4e-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-c4b57c4e-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-c4b57c4e-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-c4b57c4e-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-c4b57c4e-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| scheduled-backup | failed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | failed |
| upgrade | failed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/b28be406d7ac58c7d916e8dedf10c0f14c32188c
image: `perconalab/percona-postgresql-operator:PR-146-b28be406-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-b28be406-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-b28be406-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-b28be406-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-b28be406-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-b28be406-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | failed |
| upgrade | failed |
| users | passed |
| scheduled-backup | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/27b39207f29bb3429f1fa52f11ac72410c86e096
image: `perconalab/percona-postgresql-operator:PR-146-27b39207-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-27b39207-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-27b39207-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-27b39207-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-27b39207-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-27b39207-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | failed |
| init-deploy | failed |
| data-migration-gcs | passed |
| scaling | failed |
| scheduled-backup | failed |
| recreate | failed |
| affinity | failed |
| monitoring | failed |
| self-healing | failed |
| operator-self-healing | failed |
| clone-cluster | failed |
| tls-check | failed |
| upgrade | failed |
| users | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/e9c0d3d7ead791f687352d23f882e6a769723dfc
image: `perconalab/percona-postgresql-operator:PR-146-e9c0d3d7-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-e9c0d3d7-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-e9c0d3d7-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-e9c0d3d7-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-e9c0d3d7-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-e9c0d3d7-pgo-deployer`

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | failed |
| tls-check | failed |
| upgrade | failed |
| scheduled-backup | failed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/e9c0d3d7ead791f687352d23f882e6a769723dfc
image: `perconalab/percona-postgresql-operator:PR-146-e9c0d3d7-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-e9c0d3d7-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-e9c0d3d7-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-e9c0d3d7-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-e9c0d3d7-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-e9c0d3d7-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| scheduled-backup | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | failed |
| tls-check | passed |
| upgrade | failed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/5b56f485fa2367fbdee19e59d9c51d7210a59c97
image: `perconalab/percona-postgresql-operator:PR-146-5b56f485-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-5b56f485-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-5b56f485-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-5b56f485-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-5b56f485-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-5b56f485-pgo-deployer`

**Review by cap1984 — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| scheduled-backup | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | failed |
| tls-check | passed |
| upgrade | passed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/146/commits/0176b7096932401d0e81dc5752e7572f3113b6fb
image: `perconalab/percona-postgresql-operator:PR-146-0176b709-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-146-0176b709-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-146-0176b709-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-146-0176b709-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-146-0176b709-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-146-0176b709-pgo-deployer`

**Pull request merged**